enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,258 @@
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
+platforms, and is the subject of a [long-standing enhancement
+proposal](https://github.com/openshift/enhancements/pull/201). Other issues
+specific to the baremetal platform are that boot times for bare metal servers
+can be very long (and therefore the reboot is costly), and that support for
+particular hardware may theoretically require a particular version of CoreOS.
+
+We are changing the deploy disk image to use CoreOS. This takes the form of
+both an ISO (for hosts that can use virtualmedia) and of a kernel + initrd +
+rootfs (for hosts that use PXE). When upgrading an existing disconnected
+cluster, we have no way to acquire these images without the user manually
+intervening to mirror them.
+
+Like the QCOW2 provisioning disk image, the URLs for these images are known by
+the installer, but they point to the cloud by default and would have to be
+customised by the user at install time to allow disconnected installs. IPA
+itself is delivered separately, as a container image as part of the OpenShift
+release payload.
+
+Following a similar approach to that currently used with the QCOW2 also
+effectively extends the limitation that we are not updating the provisioning OS
+image to include the deploy image as well (although we will continue to be able
+to update IPA itself).
+
+We wish to avoid these problems by obtaining an up-to-date CoreOS from the
+release payload.
+
+### Goals
+
+* Ensure that no matter which version of OpenShift a cluster was installed
+  with, we are able to deliver updates to IPA and the OS it runs on.
+* Stop maintaining and shipping the non-CoreOS, RHEL-based IPA PXE files.
+* Never break existing clusters, even if they are deployed in disconnected
+  environments.
+* Eliminate the extra reboot required to update CoreOS after initial
+  provisioning.
+
+### Non-Goals
+
+* Automatically switch pre-existing MachineSets to deploy with `coreos-install`
+  instead of via QCOW2 images.
+* Update the CoreOS QCOW2 image in the cluster with each OpenShift release.
+* Build CoreOS images for platforms other than baremetal.
+
+## Proposal
+
+Build a container image containing the latest CoreOS ISO and the
+coreos-installer binary. This container can be used e.g. as an init container
+to make the ISO available where required. The `coreos-installer iso extract
+pxe` command can also be used to produce the kernel, initrd, and rootfs from
+the ISO, for the purposes of PXE booting.
+
+This image could either replace the existing content of the
+`ironic-ipa-downloader` repo, be built from the new
+`image-customization-controller` repo, or be built from a new repo.
```

Comment:
Personally I think it'd be clearer if we created a new repo/image than re-purposing either of these?